0000-interledger-quoting-protocol-v2.md

```diff
@@ -0,0 +1,87 @@
+
+1. The sender constructs an ILP packet with the receiver's ILP Address, an amount equal to the maximum value for unsigned 64-bit integers (18446744073709551615), and no data.
+2. The sender uses their ILP module to send the packet on a transfer with the amount set to the fixed source amount of their quote request.
+3. Connectors forward the payment to the receiver. (Note that connectors that recognize the special condition MAY "forward" the payment without actually reserving funds, since they know the payment will not be executed.)
+4. When the receiver is notified of the incoming transfer, they reject the transfer with an error message including the amount that arrived in the transfer and the amount of time they have before the transfer expires.
+5. Connectors relay the error message back to the sender.
+6. When the sender is notified that their test transfer was rejected, they inspect the error message to determine how much arrived at the receiver's ledger and how much time the receiver had to respond before the transfer would time out. The sender can then adjust the values before sending the actual payment.
+
+### Fixed Destination Amount
+
+The steps are the same as for Fixed Source Amount quotes, except that the transfer amount is set to an arbitrary probe value. When the sender gets the error response from the receiver, they compute the exchange rate based on the amount they sent and the amount that arrived. The sender then divides the fixed destination amount from the quote request by that exchange rate.
+
+## Specification
+
+### Condition
+
+ILQPv2 test payments use the condition:
+
+`fffffffffffffff8a5aa9fefdbed3fffffffffffffffffffffffffffffffffff`, encoded as hexadecimal.
+
+Or `__________ilqp_v2-0________________________`, encoded in Base64-URL.
+
+### Error Message Format
+
+**TODO: Define a binary format**
+
+Fields include:
+- Error code: either `F08` (which is currently unused) or `F99: Application Error`
+- Error name: `ILQPv2`
+- Amount that arrived in the final transfer to the receiver
+- Milliseconds between the final transfer expiry and when the receiver processed the notification
+
+## Open Questions
+
+1. Should this be a standalone protocol or should it be built into Transport Layer protocols like PSK? One advantage of building it into something like PSK is that in that case you could use the shared secret to authenticate and encrypt the quote message.
+2. Will senders want to hide the fact that they are requesting a quote from connectors (for example by using data in the ILP Packet as opposed to a well-known condition to indicate that the payment meant as a quote request)? Connectors may have good reasons (such as DoS prevention) for wanting to avoid having lots of failed payments going through them. On the other hand, using a known condition could allow connectors that recognize this protocol to avoid placing funds on hold.
```

Comment:
I don't think hiding the fact that you're quoting is necessary. It would be quite involved to truly make it impossible to tell that you're quoting. For now, it's probably good enough to detect if connectors treat quotes differently and depeer if a connector is acting badly.